facturacion_mexico/facturacion_fiscal/tests/test_get_or_create_concurrencia.py

```
@@ -0,0 +1,248 @@
+	def test_rollback_libera_lock(self):
+		si = self._si()
+
+		# Forzar excepción DESPUÉS de adquirir el for_update (al insertar el FFM).
+		with patch(
+			"facturacion_mexico.facturacion_fiscal.doctype.factura_fiscal_mexico."
+			"factura_fiscal_mexico.frappe.get_doc"
+		) as _mock:
+			# Dejamos pasar la lectura for_update real y rompemos en la creación del FFM.
+			real_get_doc = frappe.get_doc
+
+			def _side(*args, **kwargs):
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `_side`

(ANN202)